File: php/Program/class/Laptop.php
<?php
require_once "ProdukElektronik.php";

// class Laptop merupakan keturunan dari class ProdukElektronik
class Laptop extends ProdukElektronik {
    // atribut class Laptop
    private $tipeProsesor; // seperti intel atau amd
    private $kapasitasRAM; // dalam GB
    private $storage; // dalam GB


    // Constructor
    public function __construct($id = "", $nama = "", $harga = 0, $stok = 0, $gambar = "",
                                $merk = "", $garansi = 0, $model = "",
                                $tipeProsesor = "", $kapasitasRAM = 0, $storage = 0,
                                ) {
        parent::__construct($id, $nama, $harga, $stok, $gambar, $merk, $garansi, $model);
        $this->tipeProsesor = $tipeProsesor;
        $this->kapasitasRAM = $kapasitasRAM;
        $this->storage = $storage;
    }

    // Getter
    public function getTipeProsesor() { return $this->tipeProsesor; }
    public function getKapasitasRAM() { return $this->kapasitasRAM; }
    public function getStorage() { return $this->storage; }

    // Setter
    public function setTipeProsesor($tipeProsesor) { $this->tipeProsesor = $tipeProsesor; }
    public function setKapasitasRAM($kapasitasRAM) { $this->kapasitasRAM = $kapasitasRAM; }
    public function setStorage($storage) { $this->storage = $storage; }
}

File: python/Program/laptop.py
from produk_elektronik import ProdukElektronik

# class Laptop merupakan keturunan dari class ProdukElektronik
class Laptop(ProdukElektronik):
    # Constructor
    def __init__(self):
        super().__init__()
        self.__tipe_prosesor = ""
        self.__kapasitas_ram = 0
        self.__storage = 0

    # Getter
    def get_tipe_prosesor(self):
        return self.__tipe_prosesor

    def get_kapasitas_ram(self):
        return self.__kapasitas_ram
    
    def get_storage(self):
        return self.__storage
    
    # Setter
    def set_tipe_prosesor(self, tipe):
        self.__tipe_prosesor = tipe

    def set_kapasitas_ram(self, ram):
        self.__kapasitas_ram = ram

    def set_storage(self, storage):
        self.__storage = storage
